### PR TITLE
[Feature] Start/stop monitoring server based on monitoring config

### PR DESCRIPTION
**Pull request by michalpristas**
due to cloud scenario where they have disabled monitoring but expect server to be running

this needs to be in sync with fleet to distinguish `monitoring.enabled` because we don't want to monitor anything and disabled because server should be down

Fixes: https://github.com/elastic/elastic-agent/issues/2734

### CODE REVIEW

**Comment by cmacknz** [comment]
I think we need to change the agent configuration used in cloud first, and since we are doing that we should address https://github.com/elastic/elastic-agent/issues/2509 which will also need a cloud configuration change.

See https://github.com/elastic/elastic-agent/issues/2734#issuecomment-1758493579

**Comment by michalpristas** [comment]
this is dependent on work defined here: https://github.com/elastic/kibana/issues/168629
kibana/fleet will signal 3 states (`enabled`, `metrics`, `logs`)
- true, true, true: metrics and logs collection enabled
- true, false, false: metrics and logs are not collected but server is running
- false, false, false: metrics and logs are not collected, server is shutdown.

default behavior is either 1st or 3rd with very few exceptions which falls into seconds case.

**Comment by michalpristas** [comment]
all dependent work is done. bringing this alive

File: internal/pkg/agent/application/monitoring/reload/reload.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	sr.log.Debugf("Server started")
+	sr.isServerRunning = true
+}
+
+func (sr *ServerReloader) Stop() error {
+	sr.isServerRunningLock.Lock()
+	defer sr.isServerRunningLock.Unlock()
+
+	return sr.stop()
+}
```
[the remark this answers is not in the document]
Comment (michalpristas) [reply internal/pkg/agent/application/monitoring/reload/reload.go:75]: good point

**Comment by michalpristas** [comment]
failing on some unrelated tests, some of which are already addressed by @belimawr

**Pull request merged**